### PR TITLE
Show number of storage items in heartbeat log

### LINKED ISSUE #630: Log how many content items we are storing, in heartbeat

### PR DESCRIPTION
**Pull request by carver**
### What was wrong?

Fix #630

### To-Do
- [x] Clean up commit history

### CODE REVIEW

File: portalnet/src/storage.rs [hunk 1]
```diff
@@ -444,6 +451,7 @@ impl PortalStorage {
                 debug!("Error writing content ID {id_to_remove:?} to meta db. Reverted: {err:?}",);
             } else {
                 num_removed_items += 1;
+                self.metrics.decrease_entry_count();
```
Comment (jacobkaufmann) [inline portalnet/src/storage.rs:454]: I think this ought to go at the bottom of `evict` since that method is public and the container for deletion logic

Comment (carver) [reply portalnet/src/storage.rs]: Ah yes, good catch!

**Review by njgheorghita — APPROVED** [review]
LGTM! Just thinking out loud, but it seems like we've more or less adopted this architecture of `Storage` consuming `Metrics`. This will lead to a fair amount of these `self.metrics.increase_*` / `self.metrics.decrease_*` statements throughout the `Storage` code. These are fairly difficult to keep track of. To know when / where to insert them. Changes we make in `Storage` code might have consequential effects on these reported metrics that are not so obvious...

I'm not sure if this design was ever discussed, evaluated, or just came to be. 

It seems to me, like it would be cleaner if we could have `Metrics` consume `Storage`. For example, with respect to tracking the entry count, we could take advantage of the already existing `Storage::total_entry_count()` - as opposed to inc/dec a counter with every insertion / deletion. 

Again, this is more just thinking out loud than a request for changes

**Comment by pipermerriam** [comment]
When I wrote this in python forever ago, I had an API that let me fire "events" from just about anywhere and then a mirror'd API that let me consume those events from just about anywhere else.  This let me do things like:

```
# fire the event
self.events.offer_received.trigger(content_key)
self.events.offer_accepted.trigger(content_key)
self.events.offer_rejected.trigger(content_key, reason)
self.events.offer_accept_complete.trigger(content_key, payload)
self.events.offer_accept_failed.trigger(content_key, reason)

# listen elsewhere
self.events.offer_accept_completed.subscribe(|content_key, payload| {
    ...  # maybe record some metrics data...
})
```

This architecture made for a really clean way to deal with getting access to events that happen in very disparate parts of the codebase.... just something for you guys to consider.

**Comment by carver** [comment]
Yeah, I think ultimately we'll want to have both directions: emitting events *and* a way to request summaries, like `total_entry_count()`.

**Comment by carver** [comment]
I'm also 👍🏻 to move away from the indirection and emit the metrics straight into the Prometheus Gauges. There's already plenty of friction to move away from Prometheus. I don't mind choosing to just bind ourselves to using their API. If we want to switch to another metrics collector later, we can make an internal class that meets the same API.

**Pull request merged**